core/src/main/java/com/graphhopper/routing/lm/LandmarkStorage.java

```diff
@@ -0,0 +1,745 @@
+                if (res >= Double.MAX_VALUE)
+                    return Double.POSITIVE_INFINITY;
+
+                return edge.getDistance();
+            }
+        };
+
+        // later make edge base working! Not really necessary as when adding turn costs while routing we can still
+        // use the node based traversal as this is a smaller weight approximation
+        this.traversalMode = TraversalMode.NODE_BASED;
+        final String name = AbstractWeighting.weightingToFileName(weighting);
+        this.landmarkWeightDA = dir.find("landmarks_" + name);
+
+        this.landmarks = landmarks;
+        // one short per landmark and two directions => 2*2 byte
+        this.LM_ROW_LENGTH = landmarks * 4;
+        this.FROM_OFFSET = 0;
+        this.TO_OFFSET = 2;
+        this.landmarkIDs = new ArrayList<>();
+        this.subnetworkStorage = new SubnetworkStorage(dir, "landmarks_" + name);
+    }
+
+    public void setMinimumNodes(int minimumNodes) {
+        this.minimumNodes = minimumNodes;
+    }
+
+    public int getMinimumNodes() {
+        return minimumNodes;
+    }
+
+    SubnetworkStorage getSubnetworkStorage() {
+        return subnetworkStorage;
+    }
+
+    /**
+     * This weighting is used for the selection heuristic and is per default 'shortest' leading to a
+     * much better distribution of the landmarks compared to 'fastest'.
+     */
+    public void setLMSelectionWeighting(Weighting lmSelectionWeighting) {
+        this.lmSelectionWeighting = lmSelectionWeighting;
+    }
+
+    public Weighting getWeighting() {
+        return weighting;
+    }
+
+    boolean isInitialized() {
+        return initialized;
+    }
+
+    /**
+     * This method calculates the landmarks and initial weightings to & from them.
+     */
+    public void createLandmarks() {
+        if (isInitialized())
+            throw new IllegalStateException("Initialize the landmark storage once!");
+
+        // fill 'from' and 'to' weights with maximum value
+        long maxBytes = (long) graph.getNodes() * LM_ROW_LENGTH;
+        this.landmarkWeightDA.create(2000);
+        this.landmarkWeightDA.ensureCapacity(maxBytes);
+
+        for (long pointer = 0; pointer < maxBytes; pointer += 2) {
+            landmarkWeightDA.setShort(pointer, (short) SHORT_INFINITY);
+        }
+
+        // introduce a factor to store weight without loosing too much precision 
+        // AND making it compatible with weighting.calcWeight
+
+        // roughly approximate the maximum distance of the current area but limit to world wide case
+        // too small is dangerous regarding performance, e.g. for Germany at least 1500km is very important otherwise speed is at least twice as slow e.g. for just 1000km
+        BBox bounds = graph.getBounds();
+        double maxTmp = 30_000_000;
+        double distanceInMeter = !bounds.isValid() ? maxTmp : (Helper.DIST_EARTH.calcDist(bounds.maxLat, bounds.maxLon, bounds.minLat, bounds.minLon) > 50_000 ? maxTmp : 1_000_000);
```

Comment:
Could we split this line in several lines putting the results in a variable with a naming that explains what happens? It is hard to get what is happening here.